nat-lab/tests/test_events_link_state.py

```diff
@@ -0,0 +1,480 @@
+    alpha_setup_params: SetupParameters, beta_setup_params: SetupParameters
+) -> None:
+    async with AsyncExitStack() as exit_stack:
+        env = await setup_mesh_nodes(
+            exit_stack, [alpha_setup_params, beta_setup_params]
+        )
+        alpha, beta = env.nodes
+        client_alpha, client_beta = env.clients
+        connection_alpha, connection_beta = [
+            conn.connection for conn in env.connections
+        ]
+
+        async with Ping(connection_alpha, beta.ip_addresses[0]).run() as ping:
+            await testing.wait_long(ping.wait_for_next_ping())
+        async with Ping(connection_beta, alpha.ip_addresses[0]).run() as ping:
+            await testing.wait_long(ping.wait_for_next_ping())
+
+        await client_beta.stop_device()
+
+        await asyncio.sleep(1)
+
+        with pytest.raises(asyncio.TimeoutError):
+            async with Ping(connection_alpha, beta.ip_addresses[0]).run() as ping:
+                await testing.wait_normal(ping.wait_for_next_ping())
+
+        await asyncio.sleep(15)
+        alpha_events = client_beta.get_link_state_events(alpha.public_key)
+        beta_events = client_alpha.get_link_state_events(beta.public_key)
+
+        # 1 down when node is Connecting and 1 up when node is Connected
+        assert alpha_events == [LinkState.Down, LinkState.Up]
+        # beta will have 2 down events: 1 when is Connecting and 1 detected
+        assert beta_events == [LinkState.Down, LinkState.Up, LinkState.Down]
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "alpha_setup_params",
+    [
+        pytest.param(
+            SetupParameters(
+                connection_tag=ConnectionTag.DOCKER_CONE_CLIENT_1,
+                adapter_type=AdapterType.LinuxNativeWg,
+                connection_tracker_config=generate_connection_tracker_config(
+                    ConnectionTag.DOCKER_CONE_CLIENT_1,
+                    derp_1_limits=ConnectionLimits(1, 1),
+                ),
+                features=TelioFeatures(
+                    no_link_detection=NoLinkDetection(rtt_seconds=1),
+                    wireguard=long_persistent_keepalive_periods(),
+                ),
+            )
+        ),
+    ],
+)
+@pytest.mark.parametrize(
+    "beta_setup_params",
+    [
+        pytest.param(
+            SetupParameters(
+                connection_tag=ConnectionTag.DOCKER_CONE_CLIENT_2,
+                adapter_type=AdapterType.LinuxNativeWg,
+                connection_tracker_config=generate_connection_tracker_config(
+                    ConnectionTag.DOCKER_CONE_CLIENT_2,
+                    derp_1_limits=ConnectionLimits(1, 1),
+                ),
+                features=TelioFeatures(
+                    no_link_detection=NoLinkDetection(rtt_seconds=1),
+                    wireguard=long_persistent_keepalive_periods(),
+                ),
+            )
+        ),
+    ],
+)
+async def test_event_link_state_peers_exchanging_data_then_idling_then_resume(
```

Comment:
What was the fix so this testcase would pass?

Reply:
Hmmm... So at this state after communication goes silent and then continues it will keep repeating Down as far as I remember? If so we need to ticket it and fix it afterwards as it will be hit all the time in real life unless I understood something wrong.(might not be the case as libtelio has persistent keepalives enabled always I think)

Reply:
We have LLT-4753 for that. And yes, the flicker will happen if the peer will not reply in one second, the wg update interval.